php/stats.php

```diff
@@ -152,85 +174,89 @@
 (SELECT 'W' dir, a.* FROM visited_airports a ORDER BY x ASC LIMIT 1)
 SQL;
 
-$compass_labels = [
-    'N' => _("Northernmost"),
-    'S' => _("Southernmost"),
-    'E' => _("Easternmost"),
-    'W' => _("Westernmost")
-];
-
 $sth = $dbh->query($sql);
-$rows = [];
 foreach ($sth as $row) {
-    $dir = $compass_labels[$row["dir"]];
     $code = format_apcode2($row["iata"], $row["icao"]);
-    $rows[] = sprintf("%s,%s,%s,%s,%s", $dir, $code, $row["apid"], $row["x"], $row["y"]);
+    $response["extreme"][$row["dir"]] = [
+        "code" => $code,
+        "apid" => (int)$row["apid"],
+        "lat" => (float)$row["x"],
+        "lon" => (float)$row["y"],
+    ];
 }
-echo implode(":", $rows) . "\n";
 
 // Censor remaining info unless in full-public mode
 if ($public != "O") {
-    print "\n\n\n";
+    echo json_encode($response);
     exit;
 }
 
-// Classes (by number of flights and distance)
-$sql = "SELECT DISTINCT class, COUNT(*) num_flights, SUM(distance) distance
+// Classes
+$sql = <<<SQL
+    SELECT DISTINCT class, COUNT(*) segments, SUM(distance) distance
     FROM flights AS f
     WHERE $filter AND class != ''
     GROUP BY class
     ORDER BY class
-";
+SQL;
 $sth = $dbh->query($sql);
```

Comment:
We should pass `PDO::FETCH_ASSOC` as the second parameter, as we don't need the 0 based indexed columns anymore.